File: ui_react/src/services/ContactService.ts
import type Contact from "../models/Contact";
import type { Address } from "../models/Contact";
import type ListItem from "../models/ListItem";
import ApiService from "./ApiService";

export default class ContactService extends ApiService
{
    async delete(contactId: string)
    {
        await this.httpDelete<Contact>(`contact/${contactId}`);
    }

    async get(contactId: string)
    {
        return (await this.httpGet<Contact>(`contact/${contactId}`)).data;
    }

    async list()
    {
        return (await this.httpGet<ListItem[]>('contact')).data;
    }

    async save(contact: Contact)
    {
        return (await this.httpPost<Contact>('contact', contact)).data;
    }

    async searchAddress(postalCode: string)
    {
        return (await this.httpGet<Address>(`postalCode/${postalCode}`)).data;
    }
}

File: API/Services/ContactService.cs
using API.Models;
using API.Repositories.Interfaces;
using API.Services.Interfaces;
using API.Utils.Interfaces;

namespace API.Services
{
    public class ContactService(
        IContactRepository contactRepository,
        IUserContext userContext) : IContactService
    {
        public Task Delete(Contact contact)
        {
            throw new NotImplementedException();
        }

        public Task<Contact> Get(Contact contact)
        {
            throw new NotImplementedException();
        }

        public async Task<List<Contact>> List()
        {
            return await contactRepository.List(userId: userContext.CurrentUserId!);
        }

        public async Task Save(Contact contact)
        {
            Contact? savedContact = null;

            if (!string.IsNullOrEmpty(contact.Id))
            {
                savedContact = await contactRepository.Get(
                    userId: userContext.CurrentUserId!,
                    contactId: contact.Id);
            }

            contact.SetIds(savedContact);
            if (string.IsNullOrEmpty(contact.Id))
                await contactRepository.Create(contact);
            else
                await contactRepository.Update(contact);
        }
    }
}
